docs/scos/dev/upgradability-services/upgradability-guidelines/private-api-method-is-overridden-on-the-project-level.md

```diff
@@ -0,0 +1,82 @@
+
+
+## Example of code that causes an upgradability error
+
+The extended class `PyzCategoryImageEntityManager` overrides the private API core method `CategoryImageEntityManager::saveCategoryImageSet`.
+
+```php
+namespace Pyz\Zed\CategoryImage\Persistence;
+
+use Spryker\Zed\CategoryImage\Persistence\CategoryImageEntityManager as SprykerCategoryImageEntityManager;
+
+class PyzCategoryImageEntityManager extends SprykerCategoryImageEntityManager
+{
+    /**
+     * ...
+     */
+    public function saveCategoryImageSet(...): ...
+    {
+        ...
+    }
+}
+```
+
+## Example of related error in the Evaluator output
+
+```bash
+------------------------------------------------------------------------------------
+************************************************************************************************************************
+Evaluator\Business\Check\IsMethodOverridden\EntityManagerCheck
+Introduce a new custom method without usage of existing one. Override usage of the current method in all usage of public API.
+************************************************************************************************************************
+------------------------------------------------------------------------------------
+Pyz\Zed\CategoryImage\Persistence\PyzCategoryImageEntityManager
+{"name":"saveCategoryImageSet", "class":"Pyz\Zed\CategoryImage\Persistence\PyzCategoryImageEntityManager"}
+{"parentClass":"Pyz\Zed\CategoryImage\Persistence\CategoryImageEntityManager"}
+************************************************************************************************************************
+```
+
+## Resolving the Evaluator check error
+
+To resolve the error provided in the example, try the following in the provided order:
+1. Recommended: Extend the functionality using the [Configuration strategy](/docs/scos/dev/back-end-development/extending-spryker/development-strategies/development-strategies.html#configuration).
+2. Recommended: Extend the functionality using the [Plug and Play strategy](/docs/scos/dev/back-end-development/extending-spryker/development-strategies/development-strategies.html#plug-and-play).
+3. Recommended: Extend the functionality using the [Project Modules strategy](/docs/scos/dev/back-end-development/extending-spryker/development-strategies/development-strategies.html#project-modules).
+4. Not recommended: On the project level, give the Private API entities unique names. For an example, see [Example of resolving the error by copying and renaming the entities](#example-of-resolving-the-error-by-copying-and-renaming-the-entities).
+
+
+## Example of resolving the error by renaming the core entity
+
+Make the name of the Private API entity unique by adding `Pyz`.
+
+```php
+namespace Pyz\Zed\CategoryImage\Persistence;
+
+use Spryker\Zed\CategoryImage\Persistence\CategoryImageEntityManager as SprykerCategoryImageEntityManager;
+
+class PyzCategoryImageEntityManager extends SprykerCategoryImageEntityManager
```

Comment:
Why did we add a prefix to the EntityManager?

Reply:
Good point. Thank you. Will update.